src/components/Api.js

```diff
@@ -0,0 +1,104 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._url = baseUrl;
+    this._header = headers;
+  }
+
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error: ${res.status}`);
+  }
```

Comment:
It's great to have a method for processing the server response, so the logic is not duplicated